ViSearchSDK/ViSearchSDKTests/ViProductTest.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  ViProductTest.swift
+//  ViSearchSDKTests
+//
+//  Created by visenze on 19/3/21.
+//
+
+import XCTest
+@testable import ViSearchSDK
+
+class ViProductTest: XCTestCase {
+ 
+    private let RESPONSE: String = """
+    {
+       "product_id": "POMELO2-AF-SG_b28d580ccf5dfd999d1006f15f773bb371542559",
+       "main_image_url": "http://d3vhkxmeglg6u9.cloudfront.net/img/p/2/2/1/8/0/6/221806.jpg",
+       "data": {
+           "link": "https://iprice.sg/r/p/?_id=b28d580ccf5dfd999d1006f15f773bb371542559",
+           "product_name": "Skrrrrt Cropped Graphic Hoodie Light Grey",
+           "sale_price": {
+               "currency": "SGD",
+               "value": "44.0"
+           }
+       }
+    }
+    """;
+    
+    
+    override func setUp() {
+        super.setUp()
+    }
+    
+    override func tearDown() {
+        super.tearDown()
+    }
+    
+    func testParse() {
+        let result = parseMethod(text: RESPONSE)
+        
+        XCTAssertNotNil(result)
+        XCTAssertEqual(result!.productId, "POMELO2-AF-SG_b28d580ccf5dfd999d1006f15f773bb371542559")
+        XCTAssertEqual(result!.mainImageUrl, "http://d3vhkxmeglg6u9.cloudfront.net/img/p/2/2/1/8/0/6/221806.jpg")
+        
+        let link = result!.data["link"] as! String
+        XCTAssertEqual(link, "https://iprice.sg/r/p/?_id=b28d580ccf5dfd999d1006f15f773bb371542559")
+        
+        let productName = result!.data["product_name"] as! String
+        XCTAssertEqual(productName, "Skrrrrt Cropped Graphic Hoodie Light Grey")
+        
+        let price = result!.data["sale_price"] as! Dictionary<String,String>
+        
+        let currency = price["currency"]
+        XCTAssertEqual(currency, "SGD")
+        
+        let value = price["value"]
+        XCTAssertEqual(value, "44.0")
+    }
+    
+    func parseMethod(text: String) -> ViProduct? {
+        do {
+            let data = text.data(using: .utf8)!
+            let dict = try JSONSerialization.jsonObject(with: data, options: .allowFragments) as! Dictionary<String, Any>
```

Comment:
this is not the right way to test, you suppose to call the actual parse method in the response class and assert. By doing this, you are not verifying whether your implementation is correct. In fact this is a new implementation.